app/controllers/profile_controller.rb

```diff
@@ -11,4 +11,45 @@ def index
     @stories = Kaminari.paginate_array(get_stories(user))
                        .page(params[:page])
   end
+
+  def add_ban
+    ban(true)
+  end
+
+  def remove_ban
+    ban(false)
+  end
+
+  private
+
+  def ban_mailer(user, banned)
+    return unless user.any?
```

Comment:
The argument `user`, contrary to what it says, actually holds an `ActiveRecord_Relation` object which is a bit confusing imo. What do you think of calling the `ban_mailer` method with the `User` object instead? This way, `ban_mailer` doesn't need to check if it was passed a non-empty collection and it doesn't need to get the user object with the `user.first` call.

The `ban` and `ban_mailer` methods would look something like

```ruby
def ban(banned)
  return unless current_user.admin
  user = User.find_by(id: params[:user_id])
  return unless user

  user.update(banned: banned)
  ban_mailer(user, banned)
  redirect_to(
    admin_dashboard_path,
    notice_or_alert(
      user,
      banned ? 'user_banned' : 'ban_removed'
    )
  )
end

def ban_mailer(user, banned)
  if banned
    BannedMailer.add_ban_email(user).deliver_now
  else
    BannedMailer.remove_ban_email(user).deliver_now
  end
end
```

And we could go even further and move all the ban and remove ban logic into a Service object.

Thoughts?

Reply:
This is a really good call out! Thank you. I think for now we can keep it in the controller, but I'll update things so that the user object is actually the expected type.